staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery.go

```diff
@@ -117,11 +131,15 @@ func (r *versionDiscoveryHandler) ServeHTTP(w http.ResponseWriter, req *http.Req
 			StorageVersionHash: storageVersionHash,
 		})
 
-		subresources, err := apiextensionshelpers.GetSubresourcesForVersion(crd, requestedVersion)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
+		var subresources *apiextensionsv1.CustomResourceSubresources
+		for i := range crd.Spec.Versions {
+			v := crd.Spec.Versions[i]
+			if requestedVersion == v.Name {
+				subresources = v.Subresources
+				break
+			}
```

Comment:
nit: didn't we have some helpers for these kind of version getters?

Reply:
actually it was used. Wouldn't it be the same as ignoring the err?

Reply:
Yes, although for some reason I feel like ignoring an error is "bad". I can revert to how it was & ignore if you'd prefer?

Reply:
returning an error in that helper is actually bad style. Should be a bool.

Reply:
I actually have a a better way to get the subresources that doesn't involve iterating through crd.spec.versions a second time - will push an update in a few

Reply:
New commit added - PTAL